packages/wonder-blocks-labeled-field/src/components/labeled-field.tsx

```diff
@@ -27,17 +22,38 @@ type Props = {
      */
     description?: React.ReactNode;
     /**
-     * Whether this field is required to continue.
+     * Whether this field is required to to continue, or the error message to
+     * render if this field is left blank. This is passed into the field
+     * component.
+     *
+     * This can be a boolean or a string.
+     *
+     * String:
+     * Please pass in a translated string to use as the error message that will
+     * render if the user leaves this field blank. If this field is required,
+     * and a string is not passed in, a default untranslated string will render
+     * upon error.
+     * Note: The string will not be used if a `validate` prop is passed in.
+     *
+     * Example message: i18n._("A password is required to log in.")
+     *
+     * Boolean:
+     * True/false indicating whether this field is required. Please do not pass
+     * in `true` if possible - pass in the error string instead.
+     * If `true` is passed, and a `validate` prop is not passed, that means
+     * there is no corresponding message and the default untranlsated message
+     * will be used.
      */
-    required?: boolean;
+    required?: boolean | string;
     /**
-     * The message for the error element.
+     * The message for the error element. If there is a message, it will also
+     * set the `error` prop on the `field` component.
      *
      * Note: Since the error icon has an aria-label, screen readers will
      * prefix the error message with "Error:" (or the value provided to the
      * errorIconAriaLabel in the `labels` prop)
      */
-    error?: React.ReactNode;
+    errorMessage?: React.ReactNode;
```

Comment:
Renaming to `errorMessage` so it is clear it's the message. (Form components like TextField, TextArea, etc have `error` boolean props so it could be confusing if this one continuted to be named `error` when it doesn't expect a boolean)